crates/sdk/src/db_connection.rs

```diff
@@ -123,12 +131,36 @@ impl<M: SpacetimeModule> DbContextImpl<M> {
                     }
                     *ident_store = Some(identity);
                 }
-                assert_eq!(get_client_address(), addr);
+                assert_eq!(get_client_address(), address);
                 let mut inner = self.inner.lock().unwrap();
+
+                // Handle the `ids_to_names` mappings.
+                // Validate that all the reducer and table names are as expected.
+                // These are `O(n)` checks, but it is only done once per connection for a small set.
+                inner.reducer_name_to_id = M::Reducer::reducer_names()
+                    .iter()
+                    .copied()
+                    .zip(ids_to_names.reducer_ids.iter().copied())
+                    .collect();
+                let check_eq = |got: &[Arc<_>], need: &[&_]| got.iter().map(|n| &**n).eq(need.iter().copied());
+                anyhow::ensure!(
+                    check_eq(&ids_to_names.reducer_names, M::Reducer::reducer_names()),
+                    "actual and expected reducer names do not match"
+                );
+                anyhow::ensure!(
+                    check_eq(&ids_to_names.table_names, M::DbUpdate::table_names()),
+                    "actual and expected reducer names do not match"
+                );
```

Comment:
Why require this? It seems to me that the SDK should be able to function even if it's not aware of the full set of reducers or of tables. Recall that unrecognized reducers are (or at least should be; it's possible this is mis-implemented) parsed as `Event::UnknownTransaction`, so the SDK can survive a `TransactionUpdate` message for an unknown reducer. We can reasonably assume that the client will never subscribe to a table which it doesn't have bindings for, so there's no risk of `TableUpdate`s for unknown tables.

We have already seen users in the wild manually removing bindings for reducers and tables they don't intend to access from the client, which seems like a silly thing to do, but I see no reason not to support. More importantly, it should be possible to use outdated codegen to connect to a newer module so long as the module's schema has not changed incompatibly, i.e. it should be possible to add new tables or reducers to a module without distributing a new client. Our customers depend on both of these abilities.